app/src/main/java/com/example/weather/MainActivity.java

```diff
@@ -2,13 +2,196 @@
 
+import android.net.Uri;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.MalformedInputException;
+
+import javax.net.ssl.HttpsURLConnection;
 
 public class MainActivity extends AppCompatActivity {
 
+    private TextView widgetViewWeather;
+    private EditText widgetEditCity;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        widgetViewWeather = findViewById(R.id.msg_temperature);
+        widgetEditCity = findViewById(R.id.msg_city);
+    }
+
+    public void onClickByCity(View view) {
+        // TODO: keep this button disabled while current request not finished
+        Log.d("myTag", "OnClick start");
+
+        final URL request;
+        try {
+            String city = widgetEditCity.getText().toString();
+            checkCityString(city);
+
+            request = buildRequestUrlWithCity(city);
+            Log.d("myTag", String.valueOf(request));
+        } catch (MalformedURLException | NullPointerException | IllegalArgumentException e) {
+            e.printStackTrace();
+            msgOnWrongCity();
+            return;
+        }
+
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                Log.d("myTag", "run started");
+                String respond = null;
+                try {
+                    respond = doRequest(request);
+                } catch (IOException | NullPointerException | IllegalArgumentException e) {
+                    e.printStackTrace();
+                    Log.d("myTag", "request Error");
+                    postMessageToViewWeather(getText(R.string.error_wrong_request).toString());
+                    return;
+                }
+
+                String temperature = null;
+                try {
+                    temperature = parseTemperature(respond);
+                } catch (NullPointerException | IllegalArgumentException | JSONException e) {
+                    postMessageToViewWeather(getText(R.string.error_wrong_city).toString());
+                    Log.d("myTag", "parseTemperature() Error");
+                    return;
+                }
+
+                String temperatureTemplate = new String((String) getText(R.string.template_temperature_message));
+                String temperatureMessage = String.format(temperatureTemplate, temperature);
+                postMessageToViewWeather(temperatureMessage);
+                Log.d("myTag", "Temp:" + temperatureMessage);
+
+                Log.d("myTag", "run finished");
```

Comment:
```suggestion
                Log.d(TAG, "run finished");
```

create private static final String TAG = MainActivity.class.getSimpleName() tag and use it everywhere in the class.

Again, names are important. Take a look at Logcat, there are a lot of log lines posted every second. If every component use names like "myTag", "myComponent" etc, it would be a mess.